prog/weaver/http.go

```diff
@@ -217,17 +234,52 @@ var dnsEntriesTemplate = defTemplate("dnsEntries", `\
 
 var ipamTemplate = defTemplate("ipamTemplate", `{{printIPAMRanges .Router .IPAM}}`)
 
+type VersionMeta struct {
+	CheckEnabled bool
+	NewVersion   string
+	NextCheckAt  time.Time
+}
+
 type WeaveStatus struct {
-	Version string
-	Router  *weave.NetworkRouterStatus `json:"Router,omitempty"`
-	IPAM    *ipam.Status               `json:"IPAM,omitempty"`
-	DNS     *nameserver.Status         `json:"DNS,omitempty"`
+	Version     string
+	VersionMeta *VersionMeta               `json:"Version,omitempty"`
```

Comment:
won't that end up with two `Version` keys in the json, i.e. one for the `Version` field and one for the `VersionMeta` field (which you've mapped to the 'Version' field)?